Add weather page

aberidze opened the pull request.

aberidze committed:
initial commit

aberidze committed:
code ViewController

aberidze committed:
finish main task

aberidze committed:
change minimum deployment

aberidze committed:
remove useless comments

aberidze committed:
Merge branch 'master' into Add-Weather-Page

razmaa reviewed (approved):
NICE!!

natikhiz commented on EcoTracker App/EcoTracker App/Scenes/WeatherPage/View/WeatherDetailsViewController.swift:
```diff
@@ -0,0 +1,234 @@
+        let stackView = UIStackView()
+        stackView.axis = .vertical
+        stackView.spacing = 16
+        stackView.clipsToBounds = true
```
ეს საჭიროა?

Jamatta reviewed (approved):
space ები გაქვს ზოგგან ზედმეტი

natikhiz commented on EcoTracker App/EcoTracker App/Scenes/WeatherPage/View/WeatherDetailsViewController.swift:
```diff
@@ -0,0 +1,234 @@
+    }()
+    
+    private let humidityLabel: UILabel = {
+        let label = UILabel()
+        label.font = UIFont(name: "Helvetica", size: 14.0)
+        label.textColor = CustomColors.white
+        return label
```
ყველა ლეიბლისთვის ერთი და იგივე კოდი გვაქვს და შეგიძლია ფუნქციაში გაიტანო ლეიბლის შექმნა

natikhiz commented on EcoTracker App/EcoTracker App/Scenes/WeatherPage/View/WeatherDetailsViewController.swift:
```diff
@@ -0,0 +1,234 @@
+    private let windStackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.spacing = 16
+        stackView.backgroundColor = CustomColors.primary.withAlphaComponent(0.1)
```
არ აქვს სტეკ ვიუს ბექგრაუნდ ვიუ

NanaJimsheleishvili replied:
აქვს სტეკისთვის background color ის დასეტვა შესაძლებელია 14.0 იდან

dsila2017 reviewed (approved):
ერთადერთი final, დანარჩენი იდეალურია

aberidze committed:
update according to collaborator reviews

Modified several files based on peer reviews.

aberidze committed:
add minimum and maximum latitude and logtitude constraints

additional check that latitude is in range (-90, 90) and longtitude is in range (-180, 180)

NanaJimsheleishvili commented on EcoTracker App/EcoTracker App/Scenes/WeatherPage/View/WeatherPageView.swift:
```diff
@@ -6,13 +6,167 @@
-        // Do any additional setup after loading the view.
+        setupUI()
+        addButtonActions()
+        viewModel.delegate = self
```
ცალკე მეთოდში

NanaJimsheleishvili commented on EcoTracker App/EcoTracker App/Scenes/WeatherPage/View/WeatherPageView.swift:
```diff
@@ -6,13 +6,167 @@
+                return
+            }
+            
+            if (lat < -90 || lat > 90) || (lon < -180 || lon > 180) {
```
ესენი viewModleს

NanaJimsheleishvili commented on EcoTracker App/EcoTracker App/Scenes/WeatherPage/View/WeatherPageView.swift:
```diff
@@ -6,13 +6,167 @@
+    func weatherDataResponseReceived(_ weatherData: WeatherForecastResponseModel) {
+        DispatchQueue.main.async {
+            let weatherDetailsVC = WeatherDetailsViewController()
+            weatherDetailsVC.setupWeatherData(for: weatherData)
```
ესეც viewModelს
